### ISSUE TITLE
Additional TO DO items

### ISSUE THREAD

**Issue by jeremymiller** [AllenInstitute/scrattch.taxonomy#34]
- [ ] Add a verbose tag to mapping to avoid printing out all the output from tree and especially hierarchical mapping.
- [ ] Make the off-target selection and subsetting components of buildPatchseqTaxonomy less complicated (e.g., for subsetting, just provide a vector, and all the relevant things in the uns are also updated).
- [ ] Remove all the stuff about taxonomy directories and taxonomy mode directories for buildPatchseqTaxonomy, as they are not needed
- [ ] Allow for a cell x gene matrix with cells as ROWS to be provided (taxonomy) <- Jeremy already updated cpm/logCPM for this, but nothing else
- [ ] Provide default method for variable gene calculation in buildTaxonomy and buildPatchseqTaxonomy
- [ ] Variable genes are not selected during buildPatchseqTaxonomy <-- which genes are used for correlation mapping in this case? It looks like it's the general variable genes, but really we should be using mode-specific genes. So we should switch AIT.anndata$var$highly_variable_genes to something like AIT.anndata$var$highly_variable_genes_[MODE]

**Comment by UCDNJJ** [comment]
Agreed variable genes should be associated with a mode. We should be consistent and do something like AIT.anndata$uns$corr$mode$highly_variable_genes.

**Comment by jeremymiller** [comment]
Here are a bunch more to-dos.  I have **NOT** checked if these are already captured in other issues.  Note that the last two items might break back-compatibility.

- [ ] Allow UMAPS to be calculated internally using default method
- [ ] Allow highly_variable_genes to be omitted—if not provided, default to binary gene set  and have an option to use cell_type_mapper gene set if setupMapMyCells() is called. _(Rephrase request: Add a function called defineHighlyVariableGenes() which can return three options: (1) binary (the top N binary genes), (2) tree (all the genes that are dendrogram markers) and (3) hierarchical (all the genes used for hierarchical mapping at any level).  If 2 or 3 are asked for but not available, will return an error.)_
- [ ] Add a unit test that the hierarchy presented is correct (e.g., it actually is a hierarchy and is in the correct order); if ordered backwards, reverse the order
- [ ] Fix the function for pulling detailed results so we don’t have to call them this way: “mapping.anno@detailed_results[[“tree”]]”
- [ ] Create a function called auto_deannotate() which removes the “_label”, “_id”, and “_color” and returns two tables: (1) a cell table with the original columns, and (2) a metadata tables with colors, ids, and labels preserved (for categorical annotations only… numeric annotations can be regenerated without it).  I think this will make translations easier.  Actually this is high priority…
- [ ] Hide the output from cell_type_mapper by default (or send to a log file)
- [ ] Rename variables to more informative names (subclass.column, class.column, feature.set).  Potentially replace with “fine_type” and “broad_type”?
- [ ] Should make the column names from cell_type_mapper more consistent (e.g., “cluster_label_Hierarchical”  “cluster_Hierarchical” and “avg_correlation.Hierarchical.cluster_label”  “avg_correlation.Hierarchical.cluster” (or more generally improve the column names and ordering in “getMappingResults”)
- [ ] Consider renaming “buildPatchseqTaxonomy” to “buildSubsetTaxonomy”, or having it as a wrapper version that does this without doing all the extra patch-seq QC stuff.

**Comment by UCDNJJ** [comment]
Place medians / stats in https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.varm.html
